src/azure-cli/azure/cli/command_modules/cdn/custom/custom_afdx.py

```diff
@@ -0,0 +1,829 @@
+
+    existing_rule.actions.append(action)
+    return client.create(resource_group_name,
+                         profile_name,
+                         rule_set_name,
+                         rule_name,
+                         rule=existing_rule)
+
+
+def remove_afd_rule_condition(client: RulesOperations, resource_group_name, profile_name,
+                              rule_set_name, rule_name, index):
+    existing_rule = client.get(resource_group_name, profile_name, rule_set_name, rule_name)
+    if len(existing_rule.conditions) > 1 and index < len(existing_rule.conditions):
+        existing_rule.conditions.pop(index)
+    else:
+        logger.warning("Invalid condition index found. This command will be skipped. Please check the rule.")
+
+    return client.create(resource_group_name,
+                         profile_name,
+                         rule_set_name,
+                         rule_name,
+                         rule=existing_rule)
+
+
+def remove_afd_rule_action(client: RulesOperations, resource_group_name, profile_name, rule_set_name, rule_name, index):
+    existing_rule = client.get(resource_group_name, profile_name, rule_set_name, rule_name)
+    if len(existing_rule.actions) > 1 and index < len(existing_rule.actions):
+        existing_rule.actions.pop(index)
+    else:
+        logger.warning("Invalid condition index found. This command will be skipped. Please check the rule.")
+
+    return client.create(resource_group_name,
+                         profile_name,
+                         rule_set_name,
+                         rule_name,
+                         rule=existing_rule)
+
+
+def list_afd_rule_condition(client: RulesOperations, resource_group_name,
+                            profile_name, rule_set_name,
+                            rule_name):
+    rule = client.get(resource_group_name, profile_name, rule_set_name, rule_name)
+    return rule.conditions
+
+
+def list_afd_rule_action(client: RulesOperations, resource_group_name,
+                         profile_name, rule_set_name,
+                         rule_name):
+    rule = client.get(resource_group_name, profile_name, rule_set_name, rule_name)
+    return rule.actions
+
+
+def create_afd_security_policy(client: SecurityPoliciesOperations,
+                               resource_group_name,
+                               profile_name,
+                               security_policy_name,
+                               domains: List[str],
+                               waf_policy: str):
+
+    if any([("/afdendpoints/" not in domain.lower() and
+             "/customdomains/" not in domain.lower()) for domain in domains]):
+        raise CLIError('Domain should either be endpoint ID or custom domain ID.')
+
+    if "/frontdoorwebapplicationfirewallpolicies/" not in waf_policy.lower():
+        raise CLIError('waf_policy should be valid Front Door WAF policy ID.')
```

Comment:
CLIError will be deprecated. Please use error types defined in this doc:
https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md#available-error-types

Reply:
Thanks a lot, will use InvalidArgumentValueError in next iteration.